sklearn/utils/estimator_checks.py

```diff
@@ -3297,12 +3296,16 @@ def param_filter(p):
                 tuple,
                 type(None),
                 type,
-                types.FunctionType,
-                joblib.Memory,
             }
+
             # Any numpy numeric such as np.int32.
             allowed_types.update(np.core.numerictypes.allTypes.values())
-            assert type(init_param.default) in allowed_types, (
+
```

Comment:
I would add a comment here along the lines of (improvements more than welcome because I am not sure how to phrase this in a nice way):

```py
# Although callables are mutable, we accept them as argument default value and
# trust that neither the implementation of the callable nor of the estimator
# changes the state of the callable
```